File: src/main/java/easy/DetectCapitalUse.java
package easy;

public class DetectCapitalUse {
  public boolean solve(String word) {
    int uppercase = 1;
    int lowercase = 1;
    if (Character.isLowerCase(word.charAt(0))) return word == word.toLowerCase();

    for (Character c : word.substring(1).toCharArray()) {
      if (Character.isUpperCase(c)) uppercase++;
      else if (Character.isLowerCase(c)) lowercase++;
    }
    return word.length() == uppercase || word.length() == lowercase;
  }
}

File: lib/easy/detect_capital_use.dart
bool detectCapitalUse(String word) {
  if (word == word.toLowerCase()) return true;
  bool capsLock = RegExp(r'^([A-Z]*)$').hasMatch(word);
  if (capsLock) return capsLock;
  for (int char in word.substring(1).codeUnits) {
    if (char < 97 || char > 122) return false;
  }
  return true;
}
